src/tests/test_magic.py

```diff
@@ -1296,3 +1296,30 @@ def test_interact_and_missing_ipywidgets_installed(ip):
             "%sql --interact my_variable SELECT * FROM author LIMIT {{my_variable}}"
         )
         assert isinstance(out.error_in_exec, ModuleNotFoundError)
+
+
+def test_loading_config(request, monkeypatch, ip, capsys):
```

Comment:
Instead of creating a single `pyproject.toml` file and testing it, we can use `parametrize` and create different files (valid, invalid, empty, etc.). This will allow us to easily test different cases. You might consider splitting this test into several tests (i.e. valid, invalid, and mixed configuration).

Reply:
Made the change. Now there are 3 functions `test_valid_loading_toml`, `test_invalid_loading_toml`, and `test_mixed_loading_toml`.